development/tools/upload-block/upload-block.sh

```diff
@@ -0,0 +1,45 @@
+#! /usr/bin/env bash
+
+set -euo pipefail
+
+TENANT="anonymous"
+
+function main() {
+  if [ "$#" -ne 1 ]; then
+    echo "Expected exactly one argument: the path to a block directory." >/dev/stderr
+    exit 1
+  fi
+
+  local BLOCK_DIR="$1"
+
+  if [ ! -d "$BLOCK_DIR" ]; then
+    echo "'$BLOCK_DIR' is not a directory." >/dev/stderr
+    exit 1
+  fi
+
+  BLOCK_ULID="$(basename "$BLOCK_DIR")"
+
+  echo "Block ULID is $BLOCK_ULID."
+  echo "Uploading no-compact marker..."
+
+  cat <<EOF | aws_with_creds s3 cp - "s3://mimir-tsdb/$TENANT/markers/$BLOCK_ULID-no-compact.json"
+  {
+    "id": "$BLOCK_ULID",
+    "version": 1,
+    "no_compact_time": $(date -u +%s),
+    "reason": "manual",
+    "details": "block uploaded for debugging purposes"
+  }
+EOF
```

Comment:
we have `markblocks` tool that could be used, if it skipped the blocks-exists check.

For correctness, this marker file should also be in the block directory itself.

Reply:
Good point, I've switched to using `markblocks` in 28f3244.